### PR TITLE
IBX-201: Refactored ExportCommand

### PR DESCRIPTION
**Pull request by ciastektk**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [IBX-201](https://issues.ibexa.co/browse/IBX-201)
| **Type**                                   | improvement
| **Target Ibexa DXP version** | `v4.0.0`
| **BC breaks**                          | no
| **Doc needed**                       | no

Since data source feature has been implemented ExportCommand needs changes to be compatible with data sources.

This PR contains changes to `ExportCommand`:

- Added new namespace
- Removed unnecessary options
- Fixed command options name
- Added basic integration test ( will be extended after refactoring Export feature )

`ParametersFactory`:  https://github.com/ezsystems/ezrecommendation-client/pull/68
`CommandInputResolver`: https://github.com/ezsystems/ezrecommendation-client/pull/69
 
#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [x] Provided automated test coverage.
- [x] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [x] Asked for a review (ping `@ezsystems/php-dev-team`).

### CODE REVIEW

**Review by alongosz — APPROVED** [review]

**Review by adamwojs — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

File: tests/bundle/Command/ExportCommandTest.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+use EzSystems\EzRecommendationClient\Exception\InvalidArgumentException;
+use EzSystems\EzRecommendationClient\Exception\MissingExportParameterException;
+
+final class ExportCommandTest extends AbstractCommandTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+    }
```
Comment (Steveb-p) [inline tests/bundle/Command/ExportCommandTest.php:19]:
```suggestion
```

File: tests/bundle/Command/ExportCommandTest.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        return 'ibexa:recommendation:run-export';
+    }
+
+    public function testThrowExceptionWhenInvalidSiteAccessIsPassed(): void
+    {
+        $siteAccess = 'undefined_siteaccess';
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage(
+            sprintf('SiteAccess %s doesn\'t exists', $siteAccess)
```
Comment (Steveb-p) [inline tests/bundle/Command/ExportCommandTest.php:31]:
```suggestion
            sprintf('SiteAccess %s doesn\'t exist', $siteAccess)
```

File: tests/bundle/Command/ExportCommandTest.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        array $parameters,
+        string $expectExceptionMessage
+    ): void {
+        $this->expectException(MissingExportParameterException::class);
+        $this->expectExceptionMessage($expectExceptionMessage);
+        $this->commandTester->execute($parameters);
+    }
+
+    /**
+     * @return array<array<array<string>|string>>.
```
Comment (Steveb-p) [inline tests/bundle/Command/ExportCommandTest.php:55]: More specifically, in the end those are two-element arrays, with an array as first and a string as second element.
```suggestion
     * @return array<array{array<string>, string}>.
```

File: tests/bundle/Command/ExportCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        );
+        $this->commandTester->execute(
+            [
+                '--siteaccess' => $siteAccess,
+            ]
+        );
+    }
+
+    /**
+     * @param array<array<array<string>|string>> $parameters
```
Comment (Steveb-p) [inline tests/bundle/Command/ExportCommandTest.php:41]: This seems to be wrong. `$parameters` is only ever `array<string, string>`.

Comment (ciastektk) [reply tests/bundle/Command/ExportCommandTest.php]: Fixed: https://github.com/ezsystems/ezrecommendation-client/pull/67/commits/dad3d112e64e54c1b19c400858a1830655185e15

**Review by Steveb-p — APPROVED** [review]

**Pull request merged**